src/AppInstallerCommonCore/Experiment.cpp

```diff
@@ -0,0 +1,109 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "winget/Experiment.h"
+#include "winget/UserSettings.h"
+#include "Experiment/Experiment.h"
+#include "AppInstallerTelemetry.h"
+
+namespace AppInstaller::Settings
+{
+    namespace
+    {
+        ExperimentState GetExperimentStateInternal(Experiment::Key key, const UserSettings& userSettings)
+        {
+            if (key == Experiment::Key::None)
+            {
+                return { true, ExperimentToggleSource::Default };
+            }
+
+            if (!GroupPolicies().IsEnabled(TogglePolicy::Policy::Experiments))
+            {
+                AICLI_LOG(Core, Info, <<
+                    "Experiment " << Experiment::GetExperiment(key).Name() <<
+                    " is disabled due to group policy: " << TogglePolicy::GetPolicy(TogglePolicy::Policy::Experiments).RegValueName());
+                return { false, ExperimentToggleSource::Policy };
+            }
+
+            auto experiments = userSettings.Get<Setting::Experiments>();
+            auto experiment = Experiment::GetExperiment(key);

```

Comment:
It looks like you did this, but only to uses *after* the existing `experiment` local was created.  Move the local earlier so that it can be truly constructed once.